src/store/reducers/weatherReducer.js

```diff
@@ -1,17 +1,13 @@
 import { actionTypes } from "../action-types";
 
-const initialState = {
-  cod: "",
-  message: 0,
-  cnt: 0,
-  list: [],
-  city: {},
-};
+const initialState = {};
 
 export default (state = initialState, action) => {
   switch (action.type) {
     case actionTypes.GET_WEATHER:
       return action.payload;
+    case actionTypes.CLEAR_WEATHER:
+      return (state = initialState);
```

Comment:
it should return pure state. right now it is assigning the reference pointer of the initial state.
return {...state, initialState}

Reply:
done